### DISCUSSION TITLE
Changing the structure of the lists file

### DISCUSSION THREAD

**Discussion by Anonymous4045** [letsbuilda/shills#8] [General]
Right now, we are using JSON to store the data. However, what if a user wants to shill multiple things? (looking at you @sexnine). From what I can tell, if we stick with JSON we'd need to make every entry be `"user": ["thing", ...]`, which seems convoluted to me. I think just having `user:thing,...` in our own structure could be better. 
```
Anonymous4045:linux
sexnine:cloudflare,sveltekit,jetbrains
```
looks a lot better to me than
```json
{
    "Anonymous4045": ["linux"],
    "sexnine": ["cloudflare", "sveltekit", "jetbrains"]
}
```

**Comment by Anonymous4045** [comment]
This just has the limitation that names can't have `:` and items can't have `,` but this doesn't really seem like a problem to me.

**Comment by sexnine** [comment]
I honestly think they should all be separated, as in the root object is an array and each element is an object with the fields name and what they be shilling.

The reason being is multiple certificates (or whatever y'all end up doing) can be made for each thing a person shills, in which case it would make most sense to do the approach I suggested.

**Reply by Anonymous4045** [reply]
~~Wait can you give an example?~~
I'm dumb, that's just the second one I showed lol.

**Reply by Anonymous4045** [reply]
The comma-separated format also would allow this though, wouldn't it?

**Reply by asterized** [reply]
i think what sexnine is talking about here is
```json
{
    "sexnine": "cloudflare",
    "sexnine": "sveltekit"
}
```

**Reply by sexnine** [reply]
That's not valid JSON.

I was proposing the following:
```json
[
  {
    "name": "sexnine",
    "shills": "Cloudflare"
  },
  {
    "name": "sexnine",
    "shills": "Sveltekit"
  },
  {
    "name": "someone else",
    "shills": "something"
  }
]
```

**Reply by Anonymous4045** [reply]
Idk why we'd want that over the much simpler one. The goal is to have users edit this file themselves, so simpler is better imo.

**Reply by sexnine** [reply]
@Anonymous4045 Well, I did give a justification, so you do know why.  Simplicity is subjective, and I would say the approach I've shown is simplest of all.

**Reply by Anonymous4045** [reply]
I don't quite understand the justification. The plan is still to make a cert for each subject, so the result is the same. The only difference is one method has less syntax for the users to mess up to make contributing easier.

**Comment by interrrp** [comment]
Awesome. Add some whitespace so it looks nicer:
```yaml
somehybrid: pathlib
sexnine: cloudflare, sveltekit, jetbrains
```
...and we have not-quite-YAML 😂

**Comment by asterized** [comment]
i think the original
```json
{
    "Anonymous4045": ["linux"],
    "sexnine": ["cloudflare", "sveltekit", "jetbrains"]
}
```
is the best imo
